rust/tests/read_delta_test.rs

```diff
@@ -527,6 +528,48 @@ async fn test_poll_table_commits() {
     assert!(matches!(peek, PeekCommit::UpToDate));
 }
 
+#[tokio::test]
+async fn test_log_buffering() {
+    let path = "./tests/data/simple_table_with_no_checkpoint";
+    let max_iter = 10;
+    let buf_size = 10;
+
+    let location = deltalake::builder::ensure_table_uri(path).unwrap();
```

Comment:
I'm on the fence about letting this API leak out here, but let's roll with it for now